pylint/pyreverse/inspector.py

```diff
@@ -178,9 +187,12 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
             baseobj.specializations = specializations
         # resolve instance attributes
         node.instance_attrs_type = collections.defaultdict(list)
+        node.aggregations_type = collections.defaultdict(list)
```

Comment:
Could we add typing here? Same goes for the line below?
Something along the lines of `defaultdict[str, list[...]]`

Reply:
I'm having a mypy issue with this; it's complaining that I can't set type for non-self atributes ( `Type cannot be declared in assignment to non-self attribute  [misc]`).  The nodes come from astroid but the dicts are set by the Linker. I don't know if there is a workaround for this (@DudeNr33 ?)

Reply:
If `mypy` complains here I fear that it won't be easy to solve this. I also dislike that some random attributes are added to the `Node` instances here at runtime. The cleanest solution would probably be to create a new (data)class that wraps `nodes.ClassDef` and declares the additional attributes needed by `pyreverse` inside its `__init__`, but that's out of scope in this PR.

Reply:
yes, I followed the functionality of the Linker to set attributes to nodes with these new attributes so I think it's a Linker problem. I'm gonna mark this as resolved and open a separate issue :+1:

Reply:
issue; https://github.com/PyCQA/pylint/issues/7850